lib/apiservers/engine/backends/cache/image_cache.go

```diff
@@ -170,22 +181,9 @@ func (ic *ICache) getImageByDigest(digest digest.Digest) *metadata.ImageConfig {
 
 // Looks up image by reference.Named
 func (ic *ICache) getImageByNamed(named reference.Named) *metadata.ImageConfig {
-
-	var config *metadata.ImageConfig
-
-	if tagged, ok := named.(reference.NamedTagged); ok {
-		taggedName := tagged.Name() + ":" + tagged.Tag()
-		config = ic.cacheByName[taggedName]
-	} else {
-		// First try just the name.
-		if config, ok = ic.cacheByName[named.Name()]; !ok {
-			// try with the default docker tag
-			taggedName := named.Name() + ":" + reference.DefaultTag
-			config = ic.cacheByName[taggedName]
-		}
-	}
-
-	return copyImageConfig(config)
+	// get the imageID from the repoCache
+	id, _ := RepositoryCache().Get(named)
+	return copyImageConfig(ic.cacheByID[prefixImageID(id)])
```

Comment:
missed the first item -- if "sha256:" is returned from prefix, then the cacheByID will return nil.  Which will result in the copyImageConfig returning without error.  So not a potential error path.